DevEdu.API/DevEdu.API/Controllers/CourseController.cs

```diff
@@ -4,15 +4,25 @@
 using System.Threading.Tasks;
 using DevEdu.API.Models.InputModels;
 using Microsoft.AspNetCore.Mvc;
+using DevEdu.DAL.Repositories;
+using AutoMapper;
+using DevEdu.DAL.Models;
 
 namespace DevEdu.API.Controllers
 {
     [ApiController]
     [Route("api/[controller]")]
     public class CourseController : Controller
     {
-        public CourseController()
+        CourseRepository _courseRepository;
```

Comment:
please register these classes as scoped services in DI-container (via interface) and inject them in constructor